src/Compilers/CSharp/Portable/Binder/AwaitableInfo.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// Internal structure containing all semantic information about an await expression.
+    /// </summary>
+    internal sealed class AwaitableInfo
+    {
+        public readonly MethodSymbol GetAwaiter;
+        public readonly PropertySymbol IsCompleted;
+        public readonly MethodSymbol GetResult;
+
+        public bool IsDynamic => GetResult is null;
+
+        internal AwaitableInfo(MethodSymbol getAwaiterMethod, PropertySymbol isCompletedProperty, MethodSymbol getResultMethod)
+        {
+            this.GetAwaiter = getAwaiterMethod;
+            this.IsCompleted = isCompletedProperty;
+            this.GetResult = getResultMethod;
+        }
+
+        internal AwaitableInfo Update(MethodSymbol newGetAwaiter, PropertySymbol newIsCompleted, MethodSymbol newGetResult)
+        {
+            if (ReferenceEquals(GetAwaiter, newGetAwaiter) && ReferenceEquals(IsCompleted, newIsCompleted) && ReferenceEquals(GetResult, newGetResult))
```

Comment:
>ReferenceEquals [](start = 16, length = 15)

ReferenceEquals is fine, but we usually use cast to object and regular ```==``` operator to compare references

Reply:
Leaving as-is if that's ok.

---
In reply to: [162693080](https://github.com/dotnet/roslyn/pull/23961#discussion_r162693080) [](ancestors = 162693080)